### PR TITLE
feat(hash): added sha256

### PR DESCRIPTION
**Pull request by fernandolguevara**

### CODE REVIEW

**Comment by dfralan** [comment]
I need it ASAP!

**Comment by lerno** [comment]
Can you add an entry to releasenotes and some simple test vector unit tests? You can copy off what's done in test/unit/stdlib/hash/sha1.c3

File: test/unit/stdlib/hash/sha256.c3 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+	char[] pw = "pass\0word";
+	char[] salt = "sa\0lt";
+	char[32] out;
+	sha256::pbkdf2(pw, salt, 4096, &out);
+
+	assert(out == x'89B69D05 16F82989 3C696226 650A8687 8C029AC1 3EE27650 9D5AE58B 6466A724');
+}
+
+fn void test_sha1_million_a()
```
Comment (lerno) [inline test/unit/stdlib/hash/sha256.c3:63]: Change the name of the test. And don't forget an entry in releasenotes.md

Comment (fernandolguevara) [reply test/unit/stdlib/hash/sha256.c3]: done 👍

**Comment by lerno** [comment]
Looks good.

**Comment by lerno** [comment]
Thank you!

**Pull request merged**